src/main/java/task_tree_UI/TaskTreeUI.java

```diff
@@ -162,9 +163,13 @@ public void mouseClicked(MouseEvent e) {
                             taskTree.getLastSelectedPathComponent();
                     if (node == null) return;
                     String nodeInfo = (String) node.getUserObject();
+                    TaskUI taskUI = (TaskUI) programWindows.getWindow(WindowManager.TASK_REFERENCE_KEY);
+                    taskUI.populateView(displayId, parseNodeTextForID(nodeInfo));
+                    programWindows.closeWindow(WindowManager.TASKTREE_REFERENCE_KEY);
```

Comment:
I believe we may have discussed the possibility of keeping the TaskTreeUI open when this was called, but for now, it's good that we stick to the convention we've been consistent with so far.